lib/streamlit/auth_util.py

```diff
@@ -172,6 +172,12 @@ def validate_auth_credentials(provider: str) -> None:
 
     provider_section = auth_section.get(provider)
 
+    if "_" in provider:
+        raise StreamlitAuthError(
+            f'Auth provider name "{provider}" contains an underscore. '
+            f"Please use a provider name without underscores."
```

Comment:
nit: to follow our new exception pattern, it would be better to use dedicated exception classes here for every individual exception + provide the provide as a parameter to the exception class. Like done here: 

https://github.com/streamlit/streamlit/blob/26614161bde27b87e8269f813136b3a0572274c0/lib/streamlit/errors.py#L185-L192

We haven't really migrated most of our exceptions to this, but I think it would be good to apply this pattern to all of the new exceptions we add. 

Not something to change in this PR, but I think it would be good to plan that in as a follow up.